infra/docker/legacy-use-target/image/computer_api.py

```diff
@@ -40,15 +40,76 @@
 
 
 async def check_program_connection() -> bool:
-    """Check if the appropriate program for the target type has an established connection."""
+    """Check if the appropriate program for the target type has an established connection and desktop is ready."""
 
     try:
+        print(f'Checking for established connections for {REMOTE_CLIENT_TYPE}')
+
         # Check for established connections for the program
         _, stdout, _ = await run(
             f'netstat -tnp | grep {REMOTE_CLIENT_TYPE}', timeout=5.0
         )
-        return 'ESTABLISHED' in stdout
+        if 'ESTABLISHED' not in stdout:
+            print(f'No established connections for {REMOTE_CLIENT_TYPE}')
+            return False
+
+        # Client-specific checks
+        if REMOTE_CLIENT_TYPE == 'rdp':
+            return await check_rdp_ready()
+        elif REMOTE_CLIENT_TYPE == 'vnc':
+            return await check_vnc_ready()
+        else:
+            print(f'{REMOTE_CLIENT_TYPE} is not supported')
+            return False
+
     except (TimeoutError, Exception):
```

Comment:
we should avoid catch alls, only the exceptions we safely can handle here (timeout, connection failed, etc)

Reply:
same with the individual check vnc and rdp functions. we might have a programming error in there and won't catch it properly